src/liballoc/collections/btree/set.rs

```diff
@@ -544,42 +603,50 @@ impl<T: Ord> BTreeSet<T> {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn is_subset(&self, other: &BTreeSet<T>) -> bool {
         // Same result as self.difference(other).next().is_none()
-        // but the 3 paths below are faster (in order: hugely, 20%, 5%).
-        if self.len() > other.len() {
-            false
-        } else if self.len() > other.len() / ITER_PERFORMANCE_TIPPING_SIZE_DIFF {
-            // Self is not much smaller than other set.
-            // Stolen from TreeMap
-            let mut x = self.iter();
-            let mut y = other.iter();
-            let mut a = x.next();
-            let mut b = y.next();
-            while a.is_some() {
-                if b.is_none() {
-                    return false;
-                }
-
-                let a1 = a.unwrap();
-                let b1 = b.unwrap();
-
-                match b1.cmp(a1) {
-                    Less => (),
-                    Greater => return false,
-                    Equal => a = x.next(),
-                }
-
-                b = y.next();
-            }
-            true
-        } else {
-            // Big difference in number of elements, or both sets are empty.
-            // Iterate the small set, searching for matches in the large set.
-            for next in self {
-                if !other.contains(next) {
-                    return false;
+        // but the code below is faster (hugely in some cases).
+        match self.len() {
+            larger if larger > other.len() => false,
```

Comment:
nitpick, but this binding could be called something else than `larger`, it would be easier to read. (There is a reason for the name, but it refers to what comes next in the code, and not before.)

Reply:
I believe getting a name right is never nitpicking. I had `larger_than_other` initially, `too_big` in the previous commit. Maybe it's better to remove all meaning and name it `x`, because it doesn't need any meaning, because ideally it wouldn't be there and the pattern would be `> other.len()` on its own.